### PR TITLE
TST: indexing/test_floats.py::TestFloatIndexers::test_scalar_non_numeric

### PR DESCRIPTION
**Pull request by simonjayhawkins**
precursor to #25750

This test has some commented out code with a pytest.raises not used as a context manager.

The goal here is primarily to remove the commented out code.

some refactoring but not enough to allow full parametrization.

### CODE REVIEW

File: pandas/tests/indexing/test_floats.py [hunk 1]
```diff
@@ -61,109 +61,115 @@ def test_scalar_error(self):
...
+                        msg = (r"cannot do (label|index) indexing"
+                               r" on {klass} with these indexers \[3\.0\] of"
+                               r" {kind}"
+                               .format(klass=type(i), kind=str(float)))
                 with pytest.raises(error, match=msg):
-                    s.loc[3.0]
+                    idxr(obj)[3.0]
 
-                # contains
-                assert 3.0 not in s
```
Comment (simonjayhawkins) [inline pandas/tests/indexing/test_floats.py]: i guess that maybe this should have been `assert 3.0 not in i`

File: pandas/tests/indexing/test_floats.py [hunk 2]
```diff
@@ -61,109 +61,115 @@ def test_scalar_error(self):
                 s.iloc[3.0] = 0
 
     @ignore_ix
-    def test_scalar_non_numeric(self):
+    @pytest.mark.parametrize('index', [
```
Comment (jreback) [inline pandas/tests/indexing/test_floats.py]: i *think* we already have a fixture for these

File: pandas/tests/indexing/test_floats.py (hunk 1 above)
Comment (jreback) [reply pandas/tests/indexing/test_floats.py]: what is i?

**Review by jreback — CHANGES_REQUESTED** [review]
quick looks seems this is not much more clear than the original. maybe split get/set will be helpful

**Comment by simonjayhawkins** [comment]
> quick looks seems this is not much more clear than the original.

i agree. the main change is to include the cases that were skipped (not in the `pytest.skip` sense but in the `pass` or commented out sense.

> maybe split get/set will be helpful

i agree that there is basically only two tests here: `idxr(obj)[3.0]` and `idxr(obj)[3.0] = 0`
however, if we are checking for consistency of error messages then having them in the same parametrised test may be an advantage. (would also eliminate need for duplicating parametrisation or creating fixtures)
would you be adverse to something like..

```
@pytest.mark.parametrize('pass', ['getting', 'setting'])
def ...
    if pass == 'getting':
        # getting
         ...
         idxr(obj)[3.0]
    else:
        # setting
        ...
        idxr(obj)[3.0] = 0
```
also note that it is relatively straightforward to remove iloc completely from this test (and most of the other tests in this module) as it is tested at the start of the module, which make sense for float indexing with iloc. I felt that was out of scope for this PR. A precursor PR to refactor iloc would certainly make the test look less cluttered.

**Comment by jreback** [comment]
> also note that it is relatively straightforward to remove iloc completely from this test (and most of the other tests in this module) as it is tested at the start of the module, which make sense for float indexing with iloc. I felt that was out of scope for this PR. A precursor PR to refactor iloc would certainly make the test look less cluttered.

if you don't mind that would be great.

**Comment by simonjayhawkins** [comment]
superseded by #25866

**Pull request closed without merging**